examples/example_server_forum.py

```diff
@@ -0,0 +1,155 @@
+            if in_quotes:
+                if param == None and seeker+1 < length and json_str[seeker+1] == ":":
+                    param = json_str[quote_start+1:seeker]
+                else:
+                    return_dict[param] = json_str[quote_start+1:seeker]
+                    param = None
+                quote_start = None
+            in_quotes = not in_quotes
+        
+        seeker += 1
+    # end of while loop
+    return_dict['_length'] = seeker+1 - start_idx # how many chars to seek past in order to go past this json object
+    return return_dict
+
+# https://wiki.python.org/moin/BaseHttpServer
+class MyHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        print("")
+        print("GET for", self.client_address)
+        print("Path:", self.path)
+
+        output_txt = ""
+        data_dict = {}
+        try:
+            # params is everything after "?"
+            params = self.path[self.path.index("?")+1:]
+            data_dict = urllib.parse.parse_qs(params)
+        finally:
+            print("Request params:", data_dict)
+
+        try:
+            fd = open(posts_file, "r")
+            # If no user specified, get all comments.
+            if not 'user' in data_dict:
+                output_txt = fd.read()
+            # Fetch comments from specified user
+            else:
+                comment = ""
+                cnt = 0
+                fd = open(posts_file, "r")
+                while True:
+                    comment = fd.readline()
+                    if not comment:
+                        break
+                    cmt_json = read_simple_json(comment, 0)
+                    #print(cmt_json)
+                    if(cmt_json['user'] == data_dict['user'][0]):
+                        cnt += 1
+                        output_txt += comment
+                print("Number of matches:", cnt)
+                if(cnt == 0):
+                    output_txt = "This user has no comments."
+        except:
+            output_txt += ("Error opening file \"" + posts_file + "\"\n")
+            output_txt += "Did you make sure it exists? Maybe post something first.\n"
+            print(output_txt)
+
+        self.send_response(200, message="Ok")
+        self.send_header("Content-Type", "text/plain")
+        self.end_headers()
+        self.wfile.write(output_txt.encode('utf-8'))
+        #self.wfile.write(jsonify("John", "Response").encode('utf-8'))
+        
+
+    # currently just act kinda like an echo server
+    def do_POST(self):
+        print("")
+        print("POST for", self.client_address)
+        print("Request line:", self.requestline)
+        #print(self.headers) # stuff like Host and Content-Type
+        data = self.rfile.read(int(self.headers.get('Content-Length'))).decode('utf-8')
+        print(data)
+        #print("Printed POST data")
+
+        # this handles weird url special character stuff, turns params into dict
+        data_dict = urllib.parse.parse_qs(data)
```

Comment:
I think you can avoid this if we send application/json data. I'm a little surprised that its needed for POST data, but maybe it's because of how text/plain data is encoded.

Reply:
When I do curl there are no special characters, but the "requests" library on the client side does add special characters into the POST data. It's a bit strange but that's why I ended up adding it.

Reply:
Ah ok, that would explain it. The application/x-www-form-urlencoded content type definitely urlencodes the data